plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt

```diff
@@ -266,6 +272,7 @@ class SsoAccessTokenProvider(
                 sleepWithCancellation(Duration.ofMillis(100), progressIndicator)
             } catch (e: ProcessCanceledException) {
                 future.cancel(true)
+                _authorization.set(null)
```

Comment:
probably also need to set to null on success before returning future result?

Reply:
yes nice catch